src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py

```diff
@@ -112,6 +138,38 @@ def iot_hub_policy_get(client, hub_name, policy_name, resource_group_name=None):
     return client.get_keys_for_key_name(resource_group_name, hub_name, policy_name)
 
 
+def iot_hub_policy_create(client, hub_name, policy_name, permissions, resource_group_name=None):
+    hub = iot_hub_get(client, hub_name, resource_group_name)
+    policies = iot_hub_policy_list(client, hub_name, hub.resourcegroup)
+    if _is_policy_existed(policies, policy_name):
+        raise CLIError('Policy {0} already existed.'.format(policy_name))
+    updated_policies = []
+    updated_policies.extend(policies)
+    updated_policies.append(SharedAccessSignatureAuthorizationRule(policy_name, permissions))
+    hub.properties.authorization_policies = updated_policies
+    return client.create_or_update(hub.resourcegroup, hub_name, hub, {'IF-MATCH': hub.etag})
+
+
+def iot_hub_policy_delete(client, hub_name, policy_name, resource_group_name=None):
+    hub = iot_hub_get(client, hub_name, resource_group_name)
+    policies = iot_hub_policy_list(client, hub_name, hub.resourcegroup)
+    if not _is_policy_existed(policies, policy_name):
+        raise CLIError('Policy {0} not found.'.format(policy_name))
+    updated_policies = [p for p in policies if p.key_name.lower() != policy_name.lower()]
+    hub.properties.authorization_policies = updated_policies
+    return client.create_or_update(hub.resourcegroup, hub_name, hub, {'IF-MATCH': hub.etag})
+
+
+def _is_policy_existed(policies, policy_name):
+    return any(p for p in policies if p.key_name.lower() == policy_name.lower())
```

Comment:
There is also a minor performance concern here. For searching a policy in a collection of policies, if the collection is a `set` instead of `list` the performance will be better.